src/Symfony/Bridge/Doctrine/Validator/Constraints/UniqueEntityValidator.php

```diff
@@ -152,14 +143,31 @@ public function validate($entity, Constraint $constraint)
             reset($result);
         }
 
-        /* If no entity matched the query criteria or a single entity matched,
-         * which is the same as the entity being validated, the criteria is
-         * unique.
-         */
-        if (0 === count($result) || (1 === count($result) && $entity === ($result instanceof \Iterator ? $result->current() : current($result)))) {
+        if (0 === count($result)) {
             return;
         }
 
+        if (1 === count($result)) {
+            $match = $result instanceof \Iterator ? $result->current() : current($result);
+
+            if ($object === $match) {
+                return;
+            }
+
+            $method = $constraint->isEntityToUpdateMethod;
+            if (null !== $method) {
+                if (!method_exists($object, $method)) {
+                    throw new ConstraintDefinitionException(sprintf('Method "%s" does not exist in class %s', $method, $objectClass));
+                }
+
+                $reflMethod = new \ReflectionMethod($object, $method);
+
+                if ($reflMethod->isStatic() ? $reflMethod->invoke(null, $object, $match) : $reflMethod->invoke($object, $match)) {
```

Comment:
Also we could pass here the current `$em` as argument?

Reply:
Do you have an example showing how it could be used?

Reply:
Both objects can't be compared directly, right? So I guess inside this method we need to find somehow (`$em`) the current object to compare it.

Reply:
A DTO would include the entity identifier so it would not need the entity manager.

Reply:
not sure about dto's always including an entity id or so; in my experience dto + entity live decoupled.

> Both objects can't be compared directly, right?

Dunno, depends on context i guess. So yes having `$em` might be needed :+1: 

Also calling this "ToUpdate" is opinionated. I would expect something like `entityComparator` or so.

Reply:
How can an handler update an entity if its identifier is not part of the DTO?

Reply:
because i generate a new id / use the current id from url (or so) directly ;-)

Reply:
not sure what's the better practice though, now that i think of it. It's just a decision i made beforehand, i also dont do `static Dto::fromEntity()` and such.